### PR TITLE
Parse statistics to struct

### PR DESCRIPTION
**Pull request by thijsc**
We could consider making the field names more self-explanatory and adding some docs.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -209,16 +211,19 @@ pub unsafe extern "C" fn native_stats_cb<C: Context>(
...
+
+    let statistics_json = String::from_raw_parts(json as *mut u8, json_len, json_len);
+    match serde_json::from_str(&statistics_json) {
+        Ok(stats) => (*context).stats(stats),
+        Err(e) => error!("Could not parse statistics json: {}", e)
+    }
+
     mem::forget(context);   // Do not free the context
 
     1 // librdkafka will not free the pointer
```
Comment (fede1024) [inline src/client.rs]: I think we should return a 0 here, to make librdkafka free the pointer to the original string.

File: src/statistics.rs [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    pub consumer_lag: i64,
+    pub txmsgs: i64,
+    pub txbytes: i64,
+    pub msgs: i64,
+    pub rx_ver_drops: i64
+}
+
+#[derive(Deserialize,Debug)]
+pub struct Rebalance {
+    pub rebalance_age: i64,
```
Comment (fede1024) [inline src/statistics.rs]: we already have a structure named Rebalance. can we call it differently?

Comment (fede1024) [inline src/statistics.rs]: The same actually apply to many other data structures. Let's leave it as is, it's unlikely to be a problem in practice.

File: src/client.rs (hunk 1 above)
Comment (thijsc) [inline src/client.rs]: I'm not sure what the point of this return code is, I only changed it up to line 221.

**Comment by thijsc** [comment]
@fede1024 do you think we should give the fields more human readable names?

File: src/client.rs (hunk 1 above)
Comment (fede1024) [inline src/client.rs]: Sorry, my mistake. Our rust code will take care of freeing the string in this case as well.

**Comment by fede1024** [comment]
I think that using the same naming convention that librdkafka uses has the advantage that all the librdkafka documentation automatically applies to ours (plus, the names are a bit shorter).

What we should do is to copy the documentation from the librdkafka documentation to each field in each structure, so that the meaning of each value will appear in the rust docs.

Let me know if you prefer merging the change as-is and we'll take care of the documentation later on, or if you prefer to include it in this code review.

**Comment by thijsc** [comment]
I'll add the docs too. The length of the names doesn't matter in Rust right? I'm still a bit on the fence. If we add the docs inline you wouldn't need the original docs and it would be nice to have readable field names.

**Comment by thijsc** [comment]
Tested this on our staging environment and made some fields optional. It should now work properly.

File: src/statistics.rs [hunk 3]
```diff
@@ -0,0 +1,495 @@
+use std::collections::HashMap;
+
+#[derive(Deserialize,Debug)]
+pub struct Statistics {
+    pub name: String,
+    #[serde(rename = "type")]
```
Comment (fede1024) [inline src/statistics.rs]: Why this renaming?

Comment (thijsc) [inline src/statistics.rs]: `type` is a reserved word in Rust.

Comment (fede1024) [inline src/statistics.rs]: /facepalm/ yes of course :)

**Pull request merged**